File: gluecodium/src/test/resources/smoke/enums/output/android/com/example/smoke/Enums.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
public final class Enums extends NativeBase {
    public enum SimpleEnum {
        FIRST(0),
        SECOND(1);
        public final int value;
        SimpleEnum(final int value) {
            this.value = value;
        }
    }
    public enum InternalErrorCode {
        ERROR_NONE(0),
        ERROR_FATAL(999);
        public final int value;
        InternalErrorCode(final int value) {
            this.value = value;
        }
    }
    public static final class ErrorStruct {
        @NonNull
        public Enums.InternalErrorCode type;
        @NonNull
        public String message;
        public ErrorStruct(@NonNull final Enums.InternalErrorCode type, @NonNull final String message) {
            this.type = type;
            this.message = message;
        }
    }
    /**
     * For internal use only.
     * @exclude
     */
    protected Enums(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    @NonNull
    public static native Enums.SimpleEnum methodWithEnumeration(@NonNull final Enums.SimpleEnum input);
    @NonNull
    public static native Enums.InternalErrorCode flipEnumValue(@NonNull final Enums.InternalErrorCode input);
    @NonNull
    public static native Enums.InternalErrorCode extractEnumFromStruct(@NonNull final Enums.ErrorStruct input);
    @NonNull
    public static native Enums.ErrorStruct createStructWithEnumInside(@NonNull final Enums.InternalErrorCode type, @NonNull final String message);
}

File: gluecodium/src/test/resources/smoke/enums/output/android-kotlin/com/example/smoke/Enums.kt
/*

 *
 */

@file:JvmName("Enums")

package com.example.smoke

import com.example.NativeBase

class Enums : NativeBase {

    enum class SimpleEnum(private val value: Int) {
        FIRST(0),
        SECOND(1);
    }
    enum class InternalErrorCode(private val value: Int) {
        ERROR_NONE(0),
        ERROR_FATAL(999);
    }
    class ErrorStruct {
        @JvmField var type: Enums.InternalErrorCode
        @JvmField var message: String



        constructor(type: Enums.InternalErrorCode, message: String) {
            this.type = type
            this.message = message
        }





    }



    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}







    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)


        @JvmStatic external fun methodWithEnumeration(input: Enums.SimpleEnum) : Enums.SimpleEnum


        @JvmStatic external fun flipEnumValue(input: Enums.InternalErrorCode) : Enums.InternalErrorCode


        @JvmStatic external fun extractEnumFromStruct(input: Enums.ErrorStruct) : Enums.InternalErrorCode


        @JvmStatic external fun createStructWithEnumInside(type: Enums.InternalErrorCode, message: String) : Enums.ErrorStruct
    }
}
